### PR TITLE
Show 'Add People' button for implicit admins

### PR DESCRIPTION
**Pull request by akalin-keybase**

### CODE REVIEW

File: shared/teams/teamname.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  return /^([a-zA-Z0-9][a-zA-Z0-9_]?)+$/.test(s)
+}
+
+const validTeamname = (s: string): boolean => {
+  return s.split('.').every(validTeamnamePart)
+}
+
+type Teamname = string
```
Comment (akalin-keybase) [inline shared/teams/teamname.js]: If I instead import `Teamname` from `../constants/teams`, running `yarn test` gets me:

```
  ● Test suite failed to run

    TypeError: Teams.makeState is not a function

      at Object.<anonymous> (constants/entities.js:104:13)
      at Object.<anonymous> (constants/reducer.js:7:15)
      at Object.<anonymous> (constants/selectors.js:4:1)
      at Object.<anonymous> (constants/search.js:3:16)
      at Object.<anonymous> (constants/chat.js:86:239)
      at Object.<anonymous> (constants/teams.js:3:11)
      at Object.<anonymous> (teams/teamname.js:3:1)
      at Object.<anonymous> (teams/__test__/teamname.test.js:3:15)
          at Generator.next (<anonymous>)
          at new Promise (<anonymous>)
          at Generator.next (<anonymous>)
          at <anonymous>
```

which is weird.

**Comment by akalin-keybase** [comment]
ping @keybase/react-hackers

File: shared/teams/teamname.js (hunk 1 above)
Comment (buoyad) [reply shared/teams/teamname.js]: hmm, maybe related to https://github.com/keybase/client/pull/9379?

Comment (akalin-keybase) [reply shared/teams/teamname.js]: From discussion in chat, it sounds like this stems from some import cycles. Maybe this'll go away once @chrisnojima lands his changes removing some cycles...

File: shared/teams/team/container.js [hunk 2]
```diff
@@ -128,7 +158,11 @@ export default compose(
   connect(mapStateToProps, mapDispatchToProps, mergeProps),
   lifecycle({
     componentDidMount: function() {
-      this.props._loadTeam(this.props.name)
+      const teamname = this.props.name
```
Comment (akalin-keybase) [inline shared/teams/team/container.js]: Note that now we're loading details for all ancestor teams, and not just the current team.

File: shared/teams/teamname.js (hunk 1 above)
Comment (akalin-keybase) [reply shared/teams/teamname.js]: Sadly, rebasing didn't help.

**Comment by akalin-keybase** [comment]
Rebased on master and added a few more tests.

File: shared/teams/teamname.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+// @flow
+import * as I from 'immutable'
```
Comment (chrisnojima) [inline shared/teams/teamname.js]: we should move these types of helpers into constants/team or into a folder there vs w/ the components

**Comment by MarcoPolo** [comment]
lol, sorry @chrisnojima didn't realize you were reviewing too

File: shared/teams/teamname.js (hunk 3 above)
Comment (akalin-keybase) [reply shared/teams/teamname.js]: Done, moved to constants

**Pull request merged**